lib/ukfile/include/uk/pollqueue.h

```diff
@@ -0,0 +1,593 @@
+/* SPDX-License-Identifier: BSD-3-Clause */
+/* Copyright (c) 2023, Unikraft GmbH and The Unikraft Authors.
+ * Licensed under the BSD-3-Clause License (the "License").
+ * You may not use this file except in compliance with the License.
+ */
+
+/* Multi-event poll/wait queue with update chaining support */
+
+#ifndef __UKFILE_POLLQUEUE_H__
+#define __UKFILE_POLLQUEUE_H__ 1
+
+#include <uk/config.h>
+
+#include <uk/assert.h>
+#include <uk/arch/atomic.h>
+#include <uk/rwlock.h>
+#include <uk/plat/time.h>
+#include <uk/thread.h>
+
+/* Bitmask of event flags */
+typedef unsigned int uk_pollevent;
```

Comment:
It might come in use if we ever want to have unikraft/pollqueue-internal flags that are guaranteed to never overlap with posix event flags (I actually had some of these in early implementations but managed to engineer them out).
In such a case we'd change it to `unsigned long`, and any necessary type conversion between unikraft event type and posix event type remains isolated in `posix-fdio`.

Reply:
Can you add a comment to the typedef with the justification?